File: figures/07-hash-tables/snippet-string-hash.java
    public static int asciiSum(String s) {
        int sum = 0;
        for (int i = 0; i < s.length(); i++) sum += s.charAt(i);
        return sum;
    }

    public static long stringHash(String s, int base) {
        long hash = 0;
        for (int i = 0; i < s.length(); i++) hash = hash * base + s.charAt(i);
        return hash;
    }

File: figures/07-hash-tables/snippet-string-hash.c
static long ascii_sum(const char *s) {
    long sum = 0;
    for (size_t i = 0; s[i] != '\0'; i++) sum += (unsigned char)s[i];
    return sum;
}

static long long string_hash(const char *s, long long base) {
    long long hash = 0;
    for (size_t i = 0; s[i] != '\0'; i++) hash = hash * base + (unsigned char)s[i];
    return hash;
}
